src/libraries/System.Text.Json/tests/System.Text.Json.Tests/NewtonsoftTests/JsonSerializerTests.cs

```diff
@@ -130,6 +130,14 @@ public void PreserveReferencesCallbackTest()
             p2.Spouse = p1;
             Assert.Throws<JsonException> (() => JsonSerializer.Serialize(p1));
         }
+
+        [Fact]
+        public void DeserializeStructsWithParameterlessCtors()
```

Comment:
Would it be better to add this test under src\libraries\System.Text.Json\tests\Common so that it also runs with source generators?